File: examples/js/getScan.js
const axios = require('axios')
const fs = require('fs')

const HOST = "https://api.auditbase.com/";
const scan_id = "Your scan ID here";
async function get_scan(scan_id) {
  api_key =  process.env.AUDITBASE_API_KEY;
  if (!api_key) {
    throw Error(
      "API_KEY not found.  Please set the API_KEY environment variable."
    );
  }
  webhook_url = process.env.AUDITBASE_WEBHOOK_URL || "";

  console.log("webhook_url:", webhook_url);
  route = "v1/scans/";
  url = HOST + route + scan_id;
  config = {
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${api_key}`,
    },
  };

  axios.get(url, config);
  response = await axios.get(url, config);
  results_report = response.data;

  console.log("response: ", results_report);
  console.log("num issues:", results_report?.length);
}

async function get_scan_pdf(scan_id) {
    api_key =  process.env.AUDITBASE_API_KEY;
    if (!api_key) {
      throw Error(
        "API_KEY not found.  Please set the API_KEY environment variable."
      );
    }
    webhook_url = process.env.AUDITBASE_WEBHOOK_URL || "";
  
    console.log("webhook_url:", webhook_url);
    route = "v1/scans/";
    url = HOST + route + scan_id+"?format=pdf";
    config = {
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${api_key}`,
      },
      responseType: 'arraybuffer'
    };
  
    axios.get(url, config);
    response = await axios.get(url, config);
    const outputPath = "scan_report222a.pdf";
    fs.writeFileSync(outputPath, response.data);
    //results_report = response.data;

  
    //console.log("response: ", results_report);
    //console.log("num issues:", results_report?.length);
  }

//get_scan(scan_id);
get_scan_pdf(scan_id);

File: examples/python/get_scan.py
import os
import requests

HOST = "https://api.auditbase.com/"

def get_scan(scan_id):
    api_key = os.environ.get('AUDITBASE_API_KEY')
    if not api_key:
        raise(Exception("API_KEY not found.  Please set the API_KEY environment variable."))
    
    webhook_url = os.environ.get('AUDITBASE_WEBHOOK_URL', '')

    print('webhook_url:', webhook_url)
    route = 'v1/scans/'
    url = HOST + route  + scan_id
    headers = {
        'Content-Type': 'application/json',
        'Authorization': f'Bearer {api_key}',
    }


    response = requests.get(url, headers=headers)
    results_report = response.json()

    print("response: ", results_report)
    print("num issues:", len(results_report))


def get_scan_pdf(scan_id):
    api_key = os.environ.get('AUDITBASE_API_KEY')
    if not api_key:
        raise(Exception("API_KEY not found.  Please set the API_KEY environment variable."))
    
    webhook_url = os.environ.get('AUDITBASE_WEBHOOK_URL', '')

    print('webhook_url:', webhook_url)
    route = 'v1/scans/'
    url = HOST + route  + scan_id + '?format=pdf'
    headers = {
        'Content-Type': 'application/json',
        'Authorization': f'Bearer {api_key}',
    }


    response = requests.get(url, headers=headers)
    with open('scan_result22bb.pdf', 'wb') as f:
        f.write(response.content)




if __name__ == "__main__":
    scan_id = "<Your scan id>"
    get_scan(scan_id)


